### PR TITLE
Remove `includeOptional` Flags from Various Class-Related Functions

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
Within the compiler, we have a handful of functions for determining whether something uses/sends/returns classes.
Some of these had an `includeOptional` flag which is dead now; since there's no longer any optional classes to include!

These functions were almost always called with `false` anyways. The flag was added in 493caea19fd83663eb03b4eeaf7714f99ac07b97 to fix a bug with the scripting languages.

We also had analogs of these functions in the wrapper languages. I slightly cleaned them up while I was searching around.

### CODE REVIEW

File: cpp/src/slice2swift/Gen.cpp [hunk 1]
```diff
@@ -526,7 +526,7 @@ Gen::TypesVisitor::visitExceptionStart(const ExceptionPtr& p)
     }
     out << eb;
 
-    if (p->usesClasses(false) && (!base || (base && !base->usesClasses(false))))
+    if (p->usesClasses() && (!base || (base && !base->usesClasses())))
```
Comment (pepone) [inline cpp/src/slice2swift/Gen.cpp:529]: Seems we can simplify the check

```suggestion
    if (p->usesClasses() && (!base || !base->usesClasses()))
```

Does useClasses already check base types?

File: php/src/Types.cpp [hunk 2]
```diff
@@ -1218,13 +1218,11 @@ IcePHP::StructInfo::optionalFormat() const
 bool
 IcePHP::StructInfo::usesClasses() const
 {
-    for (DataMemberList::const_iterator p = members.begin(); p != members.end(); ++p)
-    {
-        if ((*p)->type->usesClasses())
+    for (const auto& dm : members)
```
Comment (pepone) [inline php/src/Types.cpp:1221]: we always write for loops with curly braces in C++

**Review by pepone — APPROVED** [review]
Looks good, see comments.

File: php/src/Types.cpp (hunk 2 above)
Comment (InsertCreativityHere) [reply php/src/Types.cpp:1221]: Yes, a typo on my part.

File: cpp/src/slice2swift/Gen.cpp (hunk 1 above)
Comment (InsertCreativityHere) [reply cpp/src/slice2swift/Gen.cpp:529]: > Does useClasses already check base types?

Yes, but after staring at this for a little while, I think it's checking something different.
This `if` condition only passes if:
- this exception uses classes
- and it's base _**doesn't**_ use classes (or it doesn't have a base)

File: cpp/src/slice2cs/Gen.cpp [hunk 3]
```diff
@@ -2086,7 +2086,7 @@ Slice::Gen::TypesVisitor::visitExceptionEnd(const ExceptionPtr& p)
     }
     _out << eb;
 
-    if ((!base || (base && !base->usesClasses(false))) && p->usesClasses(false))
+    if ((!base || (base && !base->usesClasses())) && p->usesClasses())
```
Comment (InsertCreativityHere) [inline cpp/src/slice2cs/Gen.cpp:2089]:
```suggestion
    if (p->usesClasses() && !(base && base->usesClasses()))
```

File: cpp/src/slice2js/Gen.cpp [hunk 4]
```diff
@@ -1825,7 +1825,7 @@ Slice::Gen::TypesVisitor::visitExceptionStart(const ExceptionPtr& p)
         _out << eb;
     }
 
-    if (p->usesClasses(false) && (!base || (base && !base->usesClasses(false))))
+    if (p->usesClasses() && (!base || (base && !base->usesClasses())))
```
Comment (InsertCreativityHere) [inline cpp/src/slice2js/Gen.cpp:1828]:
```suggestion
    if (p->usesClasses() && !(base && base->usesClasses()))
```

File: cpp/src/slice2swift/Gen.cpp (hunk 1 above)
Comment (InsertCreativityHere) [reply cpp/src/slice2swift/Gen.cpp]: Feel free to double check me here though.

**Pull request merged**